scripts/deployer.ts

```diff
@@ -0,0 +1,279 @@
+  console.log('postage address:' + postageContract.address + '\n block:' + contractData.blocks.postageStamp);
+  return contractData;
+}
+
+async function deployPriceOracle(contractData: any) {
+  const Oracle = await hre.ethers.getContractFactory('PriceOracle');
+  const priceOracleContract = await Oracle.deploy(contractData.addresses.postageStamp);
+  await priceOracleContract.deployed();
+  const { provider } = hre.network;
+  const txReceipt = await provider.send('eth_getTransactionReceipt', [priceOracleContract.deployTransaction.hash]);
+
+  contractData.blocks.priceOracle = parseInt(txReceipt.blockNumber, 16);
+  contractData.addresses.priceOracle = priceOracleContract.address;
+  console.log('oracle address:' + priceOracleContract.address + '\n block:' + contractData.blocks.priceOracle);
+  return contractData;
+}
+
+async function deployRedistribution(contractData: any) {
+  const Redistribution = await hre.ethers.getContractFactory('Redistribution');
+  const redistributionContract = await Redistribution.deploy(
+    contractData.addresses.staking,
+    contractData.addresses.postageStamp,
+    contractData.addresses.priceOracle
+  );
+  await redistributionContract.deployed();
+
+  const { provider } = hre.network;
+  const txReceipt = await provider.send('eth_getTransactionReceipt', [redistributionContract.deployTransaction.hash]);
+
+  contractData.blocks.redistribution = parseInt(txReceipt.blockNumber, 16);
+  contractData.addresses.redistribution = redistributionContract.address;
+  console.log(
+    'redistribution address:' + redistributionContract.address + '\n block:' + contractData.blocks.redistribution
+  );
+  return contractData;
+}
+
+async function logResult(contractData: any) {
+  console.log('\n---Contract Addresses with Blocks | Timestamp: ' + getCurrentTimestamp());
+  console.log('staking address:' + contractData.addresses.staking + '\n block:' + contractData.blocks.staking);
+  console.log(
+    'redistribution address:' +
+      contractData.addresses.redistribution +
+      '\n block:' +
+      (contractData.blocks.redistribution as number)
+  );
+  console.log('oracle address:' + contractData.addresses.priceOracle + '\n block:' + contractData.blocks.priceOracle);
+  console.log(
+    'postage address:' +
+      contractData.addresses.postageStamp +
+      '\n block:' +
+      (contractData.blocks.postageStamp as number)
+  );
+}
+
+async function rolesSetter(contractData: any) {
+  const [deployer] = await ethers.getSigners();
+
+  const PostageStamp = await ethers.getContractFactory('PostageStamp');
+  const StakeReg = await ethers.getContractFactory('StakeRegistry');
+
+  const contract = PostageStamp.attach(contractData.addresses.postageStamp);
+
+  let result: ContractTransaction;
+  let receipt: ContractReceipt;
+
+  const redistributorRole = contract.REDISTRIBUTOR_ROLE();
+  result = await contract.grantRole(redistributorRole, contractData.addresses.redistribution);
+  receipt = await result.wait();
```

Comment:
The `receipt` is never used: `receipt = await result.wait();` -> `await result.wait();`